
project hardware/qcom/audio-caf/
diff --git a/hardware/qcom/audio-caf/msm8660/AudioHardware.cpp b/hardware/qcom/audio-caf/msm8660/AudioHardware.cpp
index 0eb599a..4934b2c 100644
--- a/hardware/qcom/audio-caf/msm8660/AudioHardware.cpp
+++ b/hardware/qcom/audio-caf/msm8660/AudioHardware.cpp
@@ -4497,8 +4497,7 @@ void* AudioHardware::AudioSessionOutLPA::memBufferAlloc(int nSize, int32_t *ion_
 
     alloc_data.len =   nSize;
     alloc_data.align = 0x1000;
-    alloc_data.heap_mask = ION_HEAP(ION_AUDIO_HEAP_ID);
-    alloc_data.flags = 0;
+    alloc_data.flags = ION_HEAP(ION_AUDIO_HEAP_ID);
     int rc = ioctl(ionfd, ION_IOC_ALLOC, &alloc_data);
     if (rc) {
         ALOGE("ION_IOC_ALLOC ioctl failed\n");
