
project packages/apps/Settings/
diff --git a/packages/apps/Settings/src/com/android/settings/cyanogenmod/Processor.java b/packages/apps/Settings/src/com/android/settings/cyanogenmod/Processor.java
index 6f21e30..4e2b766 100644
--- a/packages/apps/Settings/src/com/android/settings/cyanogenmod/Processor.java
+++ b/packages/apps/Settings/src/com/android/settings/cyanogenmod/Processor.java
@@ -40,11 +40,14 @@ public class Processor extends SettingsPreferenceFragment implements
     public static final String GOV_PREF = "pref_cpu_gov";
     public static final String GOV_LIST_FILE = "/sys/devices/system/cpu/cpu0/cpufreq/scaling_available_governors";
     public static final String GOV_FILE = "/sys/devices/system/cpu/cpu0/cpufreq/scaling_governor";
+    public static final String GOV_FILE1 = "/sys/devices/system/cpu/cpu1/cpufreq/scaling_governor";
     public static final String FREQ_MIN_PREF = "pref_cpu_freq_min";
     public static final String FREQ_MAX_PREF = "pref_cpu_freq_max";
     public static final String FREQ_LIST_FILE = "/sys/devices/system/cpu/cpu0/cpufreq/scaling_available_frequencies";
     public static String FREQ_MAX_FILE = null;
+    public static final String FREQ_MAX_FILE1 = "/sys/devices/system/cpu/cpu1/cpufreq/scaling_max_freq";
     public static String FREQ_MIN_FILE = null;
+    public static final String FREQ_MIN_FILE1 = "/sys/devices/system/cpu/cpu1/cpufreq/scaling_min_freq";
     public static final String SOB_PREF = "pref_cpu_set_on_boot";
 
     protected static boolean freqCapFilesInitialized = false;
@@ -228,17 +231,21 @@ public class Processor extends SettingsPreferenceFragment implements
         initFreqCapFiles();
 
         String fname = "";
+        String fname1 = "";
 
         if (newValue != null) {
             if (preference == mGovernorPref) {
                 fname = GOV_FILE;
+                fname1 = GOV_FILE1;
             } else if (preference == mMinFrequencyPref) {
                 fname = FREQ_MIN_FILE;
+                fname1 = FREQ_MIN_FILE1;
             } else if (preference == mMaxFrequencyPref) {
                 fname = FREQ_MAX_FILE;
+                fname1 = FREQ_MAX_FILE1;
             }
 
-            if (Utils.fileWriteOneLine(fname, (String) newValue)) {
+            if (Utils.fileWriteOneLine(fname, (String) newValue) && Utils.fileWriteOneLine(fname1, (String) newValue)) {
                 if (preference == mGovernorPref) {
                     mGovernorPref.setSummary(String.format(mGovernorFormat, (String) newValue));
                 } else if (preference == mMinFrequencyPref) {
