
project frameworks/base/
diff --git a/frameworks/base/core/java/android/os/storage/StorageVolume.java b/frameworks/base/core/java/android/os/storage/StorageVolume.java
index 0285cb9..9a936d8 100644
--- a/frameworks/base/core/java/android/os/storage/StorageVolume.java
+++ b/frameworks/base/core/java/android/os/storage/StorageVolume.java
@@ -216,7 +216,9 @@ public class StorageVolume implements Parcelable {
             return -1;
         }
         try {
-            return Integer.parseInt(mUuid.replace("-", ""), 16);
+            // Integer is too small for UUID, for example "FEDC-BA98"
+            Long uuid = Long.parseLong(mUuid.replace("-", ""), 16);
+            return uuid.intValue();
         } catch (NumberFormatException e) {
             return -1;
         }
diff --git a/frameworks/base/core/jni/Android.mk b/frameworks/base/core/jni/Android.mk
index 0e3ba46..879369b 100644
--- a/frameworks/base/core/jni/Android.mk
+++ b/frameworks/base/core/jni/Android.mk
@@ -60,6 +60,7 @@ LOCAL_SRC_FILES:= \
 	android_text_AndroidCharacter.cpp \
 	android_text_AndroidBidi.cpp \
 	android_os_Debug.cpp \
+	android_os_FileUtils.cpp \
 	android_os_MemoryFile.cpp \
 	android_os_MessageQueue.cpp \
 	android_os_Parcel.cpp \
diff --git a/frameworks/base/core/jni/AndroidRuntime.cpp b/frameworks/base/core/jni/AndroidRuntime.cpp
index 8a59c6a..900c1d3 100644
--- a/frameworks/base/core/jni/AndroidRuntime.cpp
+++ b/frameworks/base/core/jni/AndroidRuntime.cpp
@@ -142,6 +142,7 @@ extern int register_android_os_SystemProperties(JNIEnv *env);
 extern int register_android_os_SystemClock(JNIEnv* env);
 extern int register_android_os_Trace(JNIEnv* env);
 extern int register_android_os_FileObserver(JNIEnv *env);
+extern int register_android_os_FileUtils(JNIEnv *env);
 extern int register_android_os_UEventObserver(JNIEnv* env);
 extern int register_android_os_MemoryFile(JNIEnv* env);
 extern int register_android_net_LocalSocketImpl(JNIEnv* env);
@@ -1170,6 +1171,7 @@ static const RegJNIRec gRegJNI[] = {
     REG_JNI(register_android_database_SQLiteDebug),
     REG_JNI(register_android_os_Debug),
     REG_JNI(register_android_os_FileObserver),
+    REG_JNI(register_android_os_FileUtils),
     REG_JNI(register_android_os_MessageQueue),
     REG_JNI(register_android_os_SELinux),
     REG_JNI(register_android_os_Trace),
