
project hardware/qcom/display-legacy/
diff --git a/hardware/qcom/display-legacy/libgralloc/framebuffer.cpp b/hardware/qcom/display-legacy/libgralloc/framebuffer.cpp
index 2005540..83730e2 100644
--- a/hardware/qcom/display-legacy/libgralloc/framebuffer.cpp
+++ b/hardware/qcom/display-legacy/libgralloc/framebuffer.cpp
@@ -281,7 +281,7 @@ int mapFrameBufferLocked(struct private_module_t* module)
     uint32_t line_length = (info.xres * info.bits_per_pixel / 8);
     info.yres_virtual = (size * numberOfBuffers) / line_length;
 
-#ifndef NO_HW_VSYNC
+#if 0
     struct msmfb_metadata metadata;
 
     metadata.op = metadata_op_base_blend;
diff --git a/hardware/qcom/display-legacy/libgralloc/ionalloc.cpp b/hardware/qcom/display-legacy/libgralloc/ionalloc.cpp
index 1ebbdfd..728bd9d 100644
--- a/hardware/qcom/display-legacy/libgralloc/ionalloc.cpp
+++ b/hardware/qcom/display-legacy/libgralloc/ionalloc.cpp
@@ -82,6 +82,11 @@ int IonAlloc::alloc_buffer(alloc_data& data)
     if (err)
         return err;
 
+	/* alloc from camera */
+	if (ION_HEAP(ION_IOMMU_HEAP_ID) == data.flags) {
+		ionAllocData.flags |= ION_HEAP(ION_CAMERA_HEAP_ID);
+	}
+
     if(data.uncached) {
         // Use the sync FD to alloc and map
         // when we need uncached memory
