
project external/svox/
diff --git a/external/svox/pico/lang/all_pico_languages.mk b/external/svox/pico/lang/all_pico_languages.mk
index 2b0cd7e..0ecf48a 100644
--- a/external/svox/pico/lang/all_pico_languages.mk
+++ b/external/svox/pico/lang/all_pico_languages.mk
@@ -14,9 +14,4 @@
 
 # This file simply inherits from all the know language packs.
 
-$(call inherit-product, external/svox/pico/lang/PicoLangDeDeInSystem.mk)
-$(call inherit-product, external/svox/pico/lang/PicoLangEnGBInSystem.mk)
 $(call inherit-product, external/svox/pico/lang/PicoLangEnUsInSystem.mk)
-$(call inherit-product, external/svox/pico/lang/PicoLangEsEsInSystem.mk)
-$(call inherit-product, external/svox/pico/lang/PicoLangFrFrInSystem.mk)
-$(call inherit-product, external/svox/pico/lang/PicoLangItItInSystem.mk)
