
project packages/inputmethods/LatinIME/
diff --git a/packages/inputmethods/LatinIME/java/res/raw/main_de.dict b/packages/inputmethods/LatinIME/java/res/raw/main_de.dict
deleted file mode 100644
index 59df8b8..0000000
Binary files a/packages/inputmethods/LatinIME/java/res/raw/main_de.dict and /dev/null differ
diff --git a/packages/inputmethods/LatinIME/java/res/raw/main_es.dict b/packages/inputmethods/LatinIME/java/res/raw/main_es.dict
deleted file mode 100644
index cb17926..0000000
Binary files a/packages/inputmethods/LatinIME/java/res/raw/main_es.dict and /dev/null differ
diff --git a/packages/inputmethods/LatinIME/java/res/raw/main_fr.dict b/packages/inputmethods/LatinIME/java/res/raw/main_fr.dict
deleted file mode 100644
index 53fd8d1..0000000
Binary files a/packages/inputmethods/LatinIME/java/res/raw/main_fr.dict and /dev/null differ
diff --git a/packages/inputmethods/LatinIME/java/res/raw/main_it.dict b/packages/inputmethods/LatinIME/java/res/raw/main_it.dict
deleted file mode 100644
index 59b78d6..0000000
Binary files a/packages/inputmethods/LatinIME/java/res/raw/main_it.dict and /dev/null differ
diff --git a/packages/inputmethods/LatinIME/java/res/raw/main_pt_br.dict b/packages/inputmethods/LatinIME/java/res/raw/main_pt_br.dict
deleted file mode 100644
index 2a8af6f..0000000
Binary files a/packages/inputmethods/LatinIME/java/res/raw/main_pt_br.dict and /dev/null differ
diff --git a/packages/inputmethods/LatinIME/java/res/raw/main_ru.dict b/packages/inputmethods/LatinIME/java/res/raw/main_ru.dict
deleted file mode 100644
index 1cddb05..0000000
Binary files a/packages/inputmethods/LatinIME/java/res/raw/main_ru.dict and /dev/null differ

project vendor/cm/
diff --git a/vendor/cm/config/common.mk b/vendor/cm/config/common.mk
index 00d3ebf..1def4f0 100644
--- a/vendor/cm/config/common.mk
+++ b/vendor/cm/config/common.mk
@@ -153,7 +153,6 @@ PRODUCT_PACKAGES += \
     DSPManager \
     libcyanogen-dsp \
     audio_effects.conf \
-    CMWallpapers \
     Apollo \
     CMUpdater \
     CMFileManager \
@@ -189,7 +188,7 @@ PRODUCT_PACKAGES += \
 PRODUCT_PACKAGES += \
     rsync
 
-PRODUCT_PACKAGE_OVERLAYS += vendor/cm/overlay/dictionaries
+#PRODUCT_PACKAGE_OVERLAYS += vendor/cm/overlay/dictionaries
 PRODUCT_PACKAGE_OVERLAYS += vendor/cm/overlay/common
 
 PRODUCT_VERSION_MAJOR = 10
