
project build/
diff --git a/build/target/product/embedded.mk b/build/target/product/embedded.mk
index 33c9918..11e02ae 100644
--- a/build/target/product/embedded.mk
+++ b/build/target/product/embedded.mk
@@ -69,7 +69,6 @@ PRODUCT_PACKAGES += \
 
 # SELinux packages
 PRODUCT_PACKAGES += \
-    auditd \
     sepolicy \
     file_contexts \
     seapp_contexts \

project vendor/cm/
diff --git a/vendor/cm/config/common.mk b/vendor/cm/config/common.mk
index b83b994..04428b2 100644
--- a/vendor/cm/config/common.mk
+++ b/vendor/cm/config/common.mk
@@ -63,9 +63,6 @@ PRODUCT_PROPERTY_OVERRIDES += \
     ro.com.android.dateformat=MM-dd-yyyy \
     ro.com.android.dataroaming=false
 
-PRODUCT_PROPERTY_OVERRIDES += \
-    ro.build.selinux=1
-
 # Thank you, please drive thru!
 PRODUCT_PROPERTY_OVERRIDES += persist.sys.dun.override=0
 
