
project build/
diff --git a/build/target/product/embedded.mk b/build/target/product/embedded.mk
index 33c9918..11e02ae 100644
--- a/build/target/product/embedded.mk
+++ b/build/target/product/embedded.mk
@@ -69,7 +69,6 @@ PRODUCT_PACKAGES += \
 
 # SELinux packages
 PRODUCT_PACKAGES += \
-    auditd \
     sepolicy \
     file_contexts \
     seapp_contexts \

project vendor/cm/
diff --git a/vendor/cm/config/common.mk b/vendor/cm/config/common.mk
index 53b321a..30f8557 100644
--- a/vendor/cm/config/common.mk
+++ b/vendor/cm/config/common.mk
@@ -59,9 +59,6 @@ PRODUCT_PROPERTY_OVERRIDES += \
     ro.com.android.dateformat=MM-dd-yyyy \
     ro.com.android.dataroaming=false
 
-PRODUCT_PROPERTY_OVERRIDES += \
-    ro.build.selinux=1
-
 ifneq ($(TARGET_BUILD_VARIANT),eng)
 # Enable ADB authentication
 ADDITIONAL_DEFAULT_PROPERTIES += ro.adb.secure=1
