
project build/
diff --git a/build/target/product/full_base.mk b/build/target/product/full_base.mk
index 2353fc4..1fc4f05 100644
--- a/build/target/product/full_base.mk
+++ b/build/target/product/full_base.mk
@@ -21,18 +21,10 @@
 
 PRODUCT_PACKAGES := \
     libfwdlockengine \
-    VideoEditor \
     WAPPushManager
 
 
 PRODUCT_PACKAGES += \
-    libvideoeditor_jni \
-    libvideoeditor_core \
-    libvideoeditor_osal \
-    libvideoeditor_videofilters \
-    libvideoeditorplayer
-
-PRODUCT_PACKAGES += \
     Galaxy4 \
     HoloSpiralWallpaper \
     LiveWallpapers \
@@ -55,9 +47,6 @@ PRODUCT_LOCALES := en_US
 # Get some sounds
 $(call inherit-product-if-exists, frameworks/base/data/sounds/AllAudio.mk)
 
-# Get the TTS language packs
-$(call inherit-product-if-exists, external/svox/pico/lang/all_pico_languages.mk)
-
 # Get a list of languages.
 $(call inherit-product, $(SRC_TARGET_DIR)/product/locales_full.mk)
 

project packages/inputmethods/LatinIME/
diff --git a/packages/inputmethods/LatinIME/java/res/raw/main_de.dict b/packages/inputmethods/LatinIME/java/res/raw/main_de.dict
deleted file mode 100644
index a59f782..0000000
Binary files a/packages/inputmethods/LatinIME/java/res/raw/main_de.dict and /dev/null differ
diff --git a/packages/inputmethods/LatinIME/java/res/raw/main_es.dict b/packages/inputmethods/LatinIME/java/res/raw/main_es.dict
deleted file mode 100644
index ac15d39..0000000
Binary files a/packages/inputmethods/LatinIME/java/res/raw/main_es.dict and /dev/null differ
diff --git a/packages/inputmethods/LatinIME/java/res/raw/main_fr.dict b/packages/inputmethods/LatinIME/java/res/raw/main_fr.dict
deleted file mode 100644
index 9044c7e..0000000
Binary files a/packages/inputmethods/LatinIME/java/res/raw/main_fr.dict and /dev/null differ
diff --git a/packages/inputmethods/LatinIME/java/res/raw/main_it.dict b/packages/inputmethods/LatinIME/java/res/raw/main_it.dict
deleted file mode 100644
index e289cef..0000000
Binary files a/packages/inputmethods/LatinIME/java/res/raw/main_it.dict and /dev/null differ
diff --git a/packages/inputmethods/LatinIME/java/res/raw/main_pt_br.dict b/packages/inputmethods/LatinIME/java/res/raw/main_pt_br.dict
deleted file mode 100644
index 8c14499..0000000
Binary files a/packages/inputmethods/LatinIME/java/res/raw/main_pt_br.dict and /dev/null differ
diff --git a/packages/inputmethods/LatinIME/java/res/raw/main_ru.dict b/packages/inputmethods/LatinIME/java/res/raw/main_ru.dict
deleted file mode 100644
index 7074416..0000000
Binary files a/packages/inputmethods/LatinIME/java/res/raw/main_ru.dict and /dev/null differ
diff --git a/packages/inputmethods/LatinIME/native/jni/Android.mk b/packages/inputmethods/LatinIME/native/jni/Android.mk
index cbe9515..cd93a9e 100644
--- a/packages/inputmethods/LatinIME/native/jni/Android.mk
+++ b/packages/inputmethods/LatinIME/native/jni/Android.mk
@@ -32,7 +32,7 @@ LOCAL_CFLAGS += -Werror -Wall -Wextra -Weffc++ -Wformat=2 -Wcast-qual -Wcast-ali
 
 ifeq ($(TARGET_ARCH), arm)
 ifeq ($(TARGET_GCC_VERSION), 4.6)
-LOCAL_CFLAGS += -Winline
+#LOCAL_CFLAGS += -Winline
 endif # TARGET_GCC_VERSION
 endif # TARGET_ARCH
 

project vendor/cm/
diff --git a/vendor/cm/config/common.mk b/vendor/cm/config/common.mk
index a204f9b..048f40b 100644
--- a/vendor/cm/config/common.mk
+++ b/vendor/cm/config/common.mk
@@ -143,9 +143,8 @@ include vendor/cm/config/themes_common.mk
 
 # Required CM packages
 PRODUCT_PACKAGES += \
-    Focal \
+    Camera \
     Development \
-    LatinIME \
     Superuser \
     su
 
@@ -162,9 +161,7 @@ PRODUCT_PACKAGES += \
     DSPManager \
     libcyanogen-dsp \
     audio_effects.conf \
-    CMWallpapers \
     Apollo \
-    CMUpdater \
     CMFileManager \
     LockClock \
     CMAccount
@@ -212,7 +209,7 @@ PRODUCT_PACKAGES += \
 # easy way to extend to add more packages
 -include vendor/extra/product.mk
 
-PRODUCT_PACKAGE_OVERLAYS += vendor/cm/overlay/dictionaries
+#PRODUCT_PACKAGE_OVERLAYS += vendor/cm/overlay/dictionaries
 PRODUCT_PACKAGE_OVERLAYS += vendor/cm/overlay/common
 
 PRODUCT_VERSION_MAJOR = 10
