
project build/
diff --git a/build/target/product/embedded.mk b/build/target/product/embedded.mk
index fe4ba4f..0830101 100644
--- a/build/target/product/embedded.mk
+++ b/build/target/product/embedded.mk
@@ -65,7 +65,6 @@ PRODUCT_PACKAGES += \
 
 # SELinux packages
 PRODUCT_PACKAGES += \
-    auditd \
     sepolicy \
     file_contexts \
     seapp_contexts \
diff --git a/build/target/product/full_base.mk b/build/target/product/full_base.mk
index 3edb864..5dd1111 100644
--- a/build/target/product/full_base.mk
+++ b/build/target/product/full_base.mk
@@ -36,7 +36,7 @@ PRODUCT_LOCALES := en_US
 $(call inherit-product-if-exists, frameworks/base/data/sounds/AllAudio.mk)
 
 # Get the TTS language packs
-$(call inherit-product-if-exists, external/svox/pico/lang/all_pico_languages.mk)
+$(call inherit-product-if-exists, external/svox/pico/lang/PicoLangEnUsInSystem.mk)
 
 # Get a list of languages.
 $(call inherit-product, $(SRC_TARGET_DIR)/product/locales_full.mk)

project packages/inputmethods/LatinIME/
diff --git a/packages/inputmethods/LatinIME/java/res/raw/main_de.dict b/packages/inputmethods/LatinIME/java/res/raw/main_de.dict
deleted file mode 100644
index a59f782..0000000
Binary files a/packages/inputmethods/LatinIME/java/res/raw/main_de.dict and /dev/null differ
diff --git a/packages/inputmethods/LatinIME/java/res/raw/main_es.dict b/packages/inputmethods/LatinIME/java/res/raw/main_es.dict
deleted file mode 100644
index ac15d39..0000000
Binary files a/packages/inputmethods/LatinIME/java/res/raw/main_es.dict and /dev/null differ
diff --git a/packages/inputmethods/LatinIME/java/res/raw/main_fr.dict b/packages/inputmethods/LatinIME/java/res/raw/main_fr.dict
deleted file mode 100644
index 9044c7e..0000000
Binary files a/packages/inputmethods/LatinIME/java/res/raw/main_fr.dict and /dev/null differ
diff --git a/packages/inputmethods/LatinIME/java/res/raw/main_it.dict b/packages/inputmethods/LatinIME/java/res/raw/main_it.dict
deleted file mode 100644
index e289cef..0000000
Binary files a/packages/inputmethods/LatinIME/java/res/raw/main_it.dict and /dev/null differ
diff --git a/packages/inputmethods/LatinIME/java/res/raw/main_pt_br.dict b/packages/inputmethods/LatinIME/java/res/raw/main_pt_br.dict
deleted file mode 100644
index 8c14499..0000000
Binary files a/packages/inputmethods/LatinIME/java/res/raw/main_pt_br.dict and /dev/null differ
diff --git a/packages/inputmethods/LatinIME/java/res/raw/main_ru.dict b/packages/inputmethods/LatinIME/java/res/raw/main_ru.dict
deleted file mode 100644
index 7074416..0000000
Binary files a/packages/inputmethods/LatinIME/java/res/raw/main_ru.dict and /dev/null differ

project vendor/cm/
diff --git a/vendor/cm/config/common.mk b/vendor/cm/config/common.mk
index babe449..9ab8c2e 100644
--- a/vendor/cm/config/common.mk
+++ b/vendor/cm/config/common.mk
@@ -70,7 +70,6 @@ PRODUCT_PROPERTY_OVERRIDES += \
     ro.com.android.dataroaming=false
 
 PRODUCT_PROPERTY_OVERRIDES += \
-    ro.build.selinux=1 \
     persist.sys.root_access=1
 
 ifneq ($(TARGET_BUILD_VARIANT),eng)
@@ -163,12 +162,9 @@ PRODUCT_PACKAGES += \
     DSPManager \
     libcyanogen-dsp \
     audio_effects.conf \
-    CMWallpapers \
     Apollo \
-    CMUpdater \
     CMFileManager \
     LockClock \
-    CMAccount
 
 # CM Hardware Abstraction Framework
 PRODUCT_PACKAGES += \
@@ -213,7 +209,7 @@ PRODUCT_PACKAGES += \
 # easy way to extend to add more packages
 -include vendor/extra/product.mk
 
-PRODUCT_PACKAGE_OVERLAYS += vendor/cm/overlay/dictionaries
+#PRODUCT_PACKAGE_OVERLAYS += vendor/cm/overlay/dictionaries
 PRODUCT_PACKAGE_OVERLAYS += vendor/cm/overlay/common
 
 PRODUCT_VERSION_MAJOR = 10
